No path was found matching log_posterior

The following test script:

https://github.com/Jammy2211/autofit_workspace_test/blob/main/multi/no_free_parameter.py

Produces the following bug:

```
Traceback (most recent call last):
  File "/mnt/c/Users/Jammy/Code/PyAuto/autofit_workspace_test/multi/no_free_parameter.py", line 91, in <module>
    instance = sample.instance_for_model(model=samples.model)
  File "/mnt/c/Users/Jammy/Code/PyAuto/PyAutoFit/autofit/non_linear/samples/sample.py", line 177, in instance_for_model
    return model.instance_from_prior_name_arguments(self.kwargs)
  File "/mnt/c/Users/Jammy/Code/PyAuto/PyAutoFit/autofit/mapper/prior_model/abstract.py", line 1331, in instance_from_prior_name_arguments
    {
  File "/mnt/c/Users/Jammy/Code/PyAuto/PyAutoFit/autofit/mapper/prior_model/abstract.py", line 1332, in <dictcomp>
    self.path_for_name(name): value
  File "/mnt/c/Users/Jammy/Code/PyAuto/PyAutoFit/autofit/mapper/prior_model/abstract.py", line 1310, in path_for_name
    raise AssertionError(f"No path was found matching {name}")
AssertionError: No path was found matching log_posterior
```

The problem is that in `sample` the `self.kwargs` is having `log_posterior` attached to it:

```
class Sample:
    def __init__(
        self, log_likelihood: float, log_prior: float, weight: float, kwargs=None
    ):
        """
        One sample taken during a search

        Parameters
        ----------
        log_likelihood
            The likelihood associated with this instance
        log_prior
            A logarithmic prior of the instance
        weight
            The weight this sample contributes to the PDF.
        kwargs
            Dictionary mapping model paths to values for the sample
        """
        self.log_likelihood = log_likelihood
        self.log_prior = log_prior
        self.weight = weight
        self.kwargs = {
            tuple(key.split(".")) if isinstance(key, str) and "." in key else key: value
            for key, value in (kwargs or dict()).items()
        }

        print(self.kwargs)
```

The print gives:

`{'gaussian_centre': 89.11417252003176, 'gaussian_normalization': 679678.0867080048, 'gaussian_sigma': 4.10753429277749, 'log_posterior': -19825761523534.39}`

I have put a hack in to fix this for now:

```
        try:
            self.kwargs.pop("log_posterior")
        except KeyError:
            pass
```

--- a/autofit/non_linear/samples/sample.py
+++ b/autofit/non_linear/samples/sample.py
@@ -1,4 +1,5 @@
 import csv
+import inspect
 from copy import copy
 from typing import List, Tuple
 
@@ -32,12 +33,7 @@ class Sample:
             for key, value in (kwargs or dict()).items()
         }
 
-        # TODO : Understand issue https://github.com/rhayes777/PyAutoFit/issues/780
-        try:
-            self.kwargs.pop("log_posterior")
-        except KeyError:
-            pass
-
+        self.kwargs.pop("log_posterior")
 
     def model_dict(self) -> dict:
         """
@@ -250,6 +246,7 @@ def load_from_table(filename: str) -> List[Sample]:
 
     sample_args = (
         "log_likelihood",
+        "log_posterior",
         "log_prior",
         "weight",
     )
@@ -263,7 +260,11 @@ def load_from_table(filename: str) -> List[Sample]:
 
             samples.append(
                 Sample(
-                    **{key: value for key, value in d.items() if key in sample_args},
+                    **{
+                        key: value
+                        for key, value in d.items()
+                        if key in inspect.getfullargspec(Sample.__init__).args
+                    },
                     kwargs={
                         key: value for key, value in d.items() if key not in sample_args
                     },

--- a/autofit/non_linear/samples/sample.py
+++ b/autofit/non_linear/samples/sample.py
@@ -33,8 +33,6 @@ class Sample:
             for key, value in (kwargs or dict()).items()
         }
 
-        self.kwargs.pop("log_posterior")
-
     def model_dict(self) -> dict:
         """
         A dictionary mapping model paths to values for the sample

